graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -189,7 +255,7 @@ private static String toStringFromJson(String name, @Nullable Object value) {
     @SuppressWarnings("unchecked")
     private static Map<String, Object> toMapFromJson(@Nullable Object maybeMap) {
         if (maybeMap == null) {
-            return ImmutableMap.of();
+            return new HashMap<>();
```

Comment:
What do you think of enforcing immutability for the return type of this method?

```
private static Map<String, Object> toMapFromJson(@Nullable Object maybeMap) {
    if (maybeMap == null) {
        return ImmutableMap.of(); <- here
    }

    if (maybeMap instanceof Map) {
        final Map<?, ?> map = (Map<?, ?>) maybeMap;
        if (map.isEmpty()) {
            return ImmutableMap.of();
        }
        return Collections.unmodifiableMap((Map<String, Object>) maybeMap); <- here
    } else {
        throw new IllegalArgumentException("Unknown parameter type variables");
    }
}
```

Reply:
This is what is missing. Thank you 👍